packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts

```diff
@@ -113,6 +118,15 @@ export function buildFoldedResultList(
   return {
     ...controller,
 
+    loadAll: (collection) =>
+      dispatch(
+        loadAll(
+          typeof collection === 'string'
+            ? collection
+            : (collection.raw[engine.state.folding.fields.collection] as string)
```

Comment:
Can we remove this ternary if the type is no longer a union? Let's simplify, either accepting the `collectionId` or the entire result, not both.

Personally, I would choose passing the result, so that developers do not have to find the collection field themselves.

Reply:
if we choose to only take one, I agree and also prefer the object.